stdlib/public/core/StringLegacy.swift

```diff
@@ -33,14 +33,28 @@ extension String {
       self = count == 0 ? "" : repeatedValue
       return
     }
+    guard !repeatedValue.isEmpty else {
+      self = ""
+      return
+    }
 
-    // TODO(String performance): We can directly call appendInPlace
-    var result = String()
-    result.reserveCapacity(repeatedValue._guts.count &* count)
-    for _ in 0..<count {
-      result += repeatedValue
+    var repeatedValue = repeatedValue
+    self = repeatedValue.withUTF8 { repeatedUTF8 in
+      let uninitializedCapacity = repeatedUTF8.count * count
+      return String(_uninitializedCapacity: uninitializedCapacity) { buffer in
+        var initialized = 0
```

Comment:
```suggestion
        if repeatedUTF8.count == 1 {
          buffer.baseAddress!.initialize(repeating: repeatedUTF8[0], count: uninitializedCapacity)
          return uninitializedCapacity
        }
        var initialized = 0
```

I checked it [in Godbolt](https://swift.godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1AKA7nioFSS%2BsgJ4BlRugDCqWgFcWDEAGYAHKROADJ4DJgAcp4ARpjEvqQADqgKhPYMrh5evgFJKXYCIWGRLDFxPlaYNvkMQgRMxAQZnt7%2BFVVptfUEhRHRsfEKdQ1NWa2DXT3Fpb4AlFao7sTI7BwApBoAglTuDMgA1ASYgxAzewC0qz7Ye7XEocB7qwDsAELrG3ufNwR3RhDEmASmCYdiMIEeACYIapVlDSHs0DsCOCuBoZu9ngARDGbTCqQ4MFICb6/B7PN6bT4Aehpe2cAJBRz2TD2YRMe0GpL2AISAKUgnuBwQmD2wDwADdGByfoKecDDughSKFEDkOY8Jh0O9qbSGH1iHtUFQDng2AoAHTavY0qlWm17ABiJD2eNYCXo8IAngsEYY9u4lEK8Ao9qFUmI8AAvWIHVAIhmHZnSrlmAgIA6MO201YANg0sIhTCiBdzGmT9xDoSTxFQJktlOtNKzVK%2Bn3oBA5jx8mJJ93%2BgPl93BBaLBfhiMEKLRzdbe15oQIEAU6Ib9tbtOUvwIIZHUSL%2B73h4PB5Lq6bZ5bZz2ynqrEwh2IChAM8%2BV7ljPQADUxO52HsACrCuWRixtyA4gvWHyNi2r4Igsk4AUBeolDGRommaoHvh2pZYZq34eJgpahgwSpgQo7i0KCwAvl8nL3JBnwJO4US0Hg%2BxhouWGyuBCr4b%2B4K3Pc47wciewAJKCKc5JWoxAJoAw%2BDVBAE7YVclw9ho8IFuEmDACCEoiiprKoB2Yi0LWmoFiuUGfMA7j1IqRmXNcXAurQgbSQ2rbWMa6lwUiXbYt2exlpcDqQhCBZ7OCuFfj%2BhFeV8AIEIsDAyY8TzYoldkOXsYBgLFfGYOawbYCwCQEJ6bkea86WfD5gURaeNlJfeqXpViOIteK9RgUCH5FY1hXxelDV%2BcNBHmqmCByP%2BDp%2BBlLx9fKmqzfNxF1Xs7b%2BgwHF4BG0YuEwCRMGqlVDTxq1zX45pGQAVP5gibclqW9n8OwKEwVCYHIu27XYB2as4x2nYQnrgjse2A0dJ1nZ6UmvHsUTuFQ30GqEm2fOYy0fmtN1Od2fmuZ5LWzkjKNo%2Bae5KBs6DoHyCj5SV/37ax0b9v1VExZd6B46sACsLz5vzmLCUiEN/eGbNAyDcPWWTrYvcQJGQyz0PA7DYOYxlWWk18PXo2r0uKn5%2BaJa2/Do8RIXmpaPjOE5tXm7O23mSYMZ%2BXgkI5g9sV47dIna2297%2BgkQIGn5bsexCOawktfvXQHSJB1tIfEIYwAin5ABKGeYBAOzIMKyAANaai88HoE%2BewQFHcRbRZZCh%2BHENh7EMzywrnwAPqNb9n3fQAsu4dQsZgLwU7EyioAusT9tTQ7k6jsQCy86dGIRIud13nzM1LUb51QNYsNznNXfN28K1Dxve3H4150nT3O58nXP8yChKA0EDXwfJuE8Fqt96HQ1qDSql9FZtWVsRIBllnavxavAl%2BmUMSZQ4HMWgnB%2Ba8G8BwLQpBUCcCEGYCwHIFhLCzhCHwPBSDIlwWguYJcQD8x8OaCEABOHMTwNA%2BHYZw7hPgNA5n0JwSQ2DNC8AIRwXgT5NK0K0HMOAsAkBoHKnQWI5BKAqISGouIphzAEF2sABABBaCelIFgcUbFMCfg1CYAA8kCHB1CaCUViE%2BCAURxGkCiKEeo4NuC8B8cwYgno7FRG0JgWwnBqEqLYIIOxDBTFeKwCwDO4g6HmLwHJOwkonwZLxJEkeKxqGzwwRk1iUR04hNcFgLxMoT4BLmFQAwwAFA2MwPYxx0TeD8EECIMQ7ApAyEEIoFQ6gMm6C4PoDOIA9EWH0HgKIT5IBzFQBVNIeSzh2OkZUSJ1RHAKRGN4KZwRQi9BQnEKZuRUgCCOXoa51RJj6j0NYPZHQhiNDcM0F5uzbDvImGcqY/QpnjGGF8rIIKPlPIuRIOYCgyHLFhcIjgWDSA4LwZIvYqg/A5jODmSQHJiEEH2IY4xpia64EIM6WEVCZi8DkfQ0gjDJA5nNFwnhHD2W8KeFMspoi0VeMkdIkAsjxEd2RRCMRGShU0LFXMSUj40ggEkEAA%3D%3D%3D) and this does actually optimise down in to a `memset`.

Very quick benchmark running `String(repeating: "x", count: 10)` 10,000 times:
- Without fast path:  ~0.001s.
- With fast path: ~0.0005s (2x faster)

Increasing `count` to 100:

- Without fast path: ~0.007s
- With fast path: ~0.002s (3.5x faster)